mainwindow.cpp

```diff
@@ -30,4 +31,30 @@ MainWindow::MainWindow(QWidget *parent) :
 MainWindow::~MainWindow()
 {
     delete ui;
+    delete canvas;
+}
+
+void MainWindow::on_actionLoad_triggered()
+{
+    jbrush::FileHandler dataHandler;  // Handles data loading for this method
+    Vector<Shape*> newShapes;   // Stores shapes loaded from file
+
+    // Bring up a panel for the user to select a directory
+    currentFile = QFileDialog::getOpenFileName(canvas, "Open", jbrush::jconstants::STARTING_DIRECTORY, "Jojibrush files (*" + jbrush::jconstants::FILE_EXTENSION + ")");
+
+    // Use the data handler to fill the shapes using data from file selected by user
+    dataHandler.loadFromFile(newShapes, currentFile);
+
+    // If some shapes were loaded, add them to the canvas and re-render the canvas
+    if(newShapes.getSize() > 0)
+    {
+        canvas->clearShapes();
+
+        for(unsigned int index = 0; index < newShapes.getSize(); ++index)
+        {
+            canvas->addShape(newShapes[index]);
```

Comment:
If using ranged-based for loops, this would just be
```suggestion
            canvas->addShape(shape);
```

Reply:
Remind me again how to do it ranged-based?

Reply:
nvm I see you put it in a different suggestion